### PR TITLE
feat: 2060 hazard mitigation

### PR DESCRIPTION
**Pull request by eoln**
- instead, two middle steps in a state machine I choose to use the barrier approach,
so the Initiate phase is blocked until both callbacks are delivering messages or timeout happens

### CODE REVIEW

**Review by kleyow — APPROVED** [review]

**Review by kleyow — COMMENTED** [review]
whoop. didn't see that tests are failing

File: src/handlers/inbound/thirdpartyRequests/transactions/{ID}.ts [hunk 1]
```diff
@@ -24,18 +24,19 @@
...
  * Handles a inbound PATCH /thirdpartyRequests/transactions/{ID} request
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 async function patch (_context: any, request: Request, h: StateResponseToolkit): Promise<ResponseObject> {
-  const payload = request.payload as ThirdpartyTransactionStatus
+  const payload = request.payload as tpAPI.Schemas.ThirdpartyRequestsTransactionsIDPatchResponse
   const pubSub = h.getPubSub()
 
-  h.getLogger().push({ payload }).info('PATCH /thirdpartyRequests/transactions/ pushing to channel')
+  h.getLogger().push({ payload }).info('PATCH /thirdpartyRequests/{ID}transactions/ pushing to channel')
```
Comment (kleyow) [inline src/handlers/inbound/thirdpartyRequests/transactions/{ID}.ts:39]:
```suggestion
  h.getLogger().push({ payload }).info('PATCH /thirdpartyRequests/{ID}/transactions/ pushing to channel')
```

**Comment by eoln** [comment]
yeap, I will fix integration after Easter.

**Review by kleyow — APPROVED** [review]
lgtm

**Pull request merged**